### DISCUSSION TITLE
Why no options for postgresql to dump privileges to schema.sql?

### DISCUSSION THREAD

**Discussion by siwang2022** [amacneil/dbmate#310] [Support]
From the code we always run pg_dump with "--no-privileges". Wonder if there is any particular reason? Thanks!
```
func (drv *Driver) DumpSchema(db *sql.DB) ([]byte, error) {
	// load schema
	args := append([]string{"--format=plain", "--encoding=UTF8", "--schema-only",
		"--no-privileges", "--no-owner"}, connectionArgsForDump(drv.databaseURL)...)
	schema, err := dbutil.RunCommand("pg_dump", args...)
	if err != nil {
		return nil, err
	}
```

**Comment by amacneil** [comment]
I don't think there is a good reason. If you can give an example where this is causing problems we could consider changing it.

**Reply by fabio-guerreiro** [reply]
Hey @amacneil 
I've encounter this "blocker" as well. In order to properly track the changes to our schema, we need the dump to include privileges granted to specific roles. For example, the following migration:
```sql
GRANT SELECT (id, first_name, last_name) ON professionals TO OWNER;
```

If afterwards someone changes the columns that the role is allowed to select, we won't be able to check the change on the schema.sql. Also, if we have that information on the schema.sql generated by the dump, we can very easily check which columns a certain role can select.
